File: LeetCode/[E] - 1207. Unique Number of Occurrences/solution.py
class Solution:
    def uniqueOccurrences(self, arr: list[int]) -> bool:
        symb_count = {}
        for num in arr:
            symb_count[num] = symb_count.get(num, 0) + 1
        count_count = {}
        for value in symb_count.values():
            if value in count_count:
                return False
            count_count[value] = 1
        return True

File: LeetCode/[E] - 1207. Unique Number of Occurrences/solution.go
package main

func uniqueOccurrences(arr []int) bool {
	symbCount := make(map[int]int, len(arr)/2)
	for _, el := range arr {
		symbCount[el] += 1
	}
	countSet := make(map[int]struct{}, len(symbCount))
	for _, value := range symbCount {
		if _, ok := countSet[value]; ok {
			return false
		}
		countSet[value] = struct{}{}
	}
	return true
}
